Source/SelfService/Web/microservice/base/view.tsx

```diff
@@ -1,66 +1,149 @@
+            '& .MuiSvgIcon-root': {
+                color: '#6678F6',
+                marginRight: theme.spacing(1),
+            },
+            '& .MuiTypography-root': {
+                color: '#6678F6',
+                textTransform: 'uppercase'
+            }
+        },
+        root: {
+            flexGrow: 1,
+        },
+        paper: {
+            padding: theme.spacing(2),
+            textAlign: 'center',
+        },
         divider: {
             backgroundColor: '#3B3D48'
         }
     })
 );
 
 export const View: React.FunctionComponent<Props> = (props) => {
+    const { enqueueSnackbar } = useSnackbar();
     const classes = useStyles();
     const $microservices = useReadable(microservices) as any;
     const history = useHistory();
+
     const _props = props!;
-    const applicationId = _props.applicationId;
+    const application = _props.application;
     const microserviceId = _props.microserviceId;
     const environment = _props.environment;
     const podsData = _props.podsData;
 
-    const [selectedKey, setSelectedKey] = useState('healthStatus');
-    // Want microservice name
-
     const currentMicroservice = $microservices.find(ms => ms.id === microserviceId);
     if (!currentMicroservice) {
-        const href = `/microservices/${applicationId}/${environment}/overview`;
+        const href = `/microservices/application/${application.id}/${environment}/overview`;
         history.push(href);
         return null;
     }
 
+    let ms = {} as MicroserviceSimple;
     let hasEditData = false;
-    if (currentMicroservice.edit &&
-        currentMicroservice.edit.dolittle &&
+    if (currentMicroservice?.edit?.dolittle &&
         currentMicroservice.id !== '' &&
         currentMicroservice.kind !== '') {
         hasEditData = true;
+        ms = currentMicroservice.edit;
+    }
+    console.log(currentMicroservice.live);
+
+    if (!hasEditData) {
+        // Can I not move this to the store?
+        let headImage = 'n/a';
+        try {
+            headImage = currentMicroservice.live.images.find(img => img.name === 'head').image;
+        } catch (e) {
+            // Intentional skip
+        }
+
+        let runtimeImage = 'n/a';
+        try {
+            runtimeImage = currentMicroservice.live.images.find(img => img.name === 'runtime').image;
+        } catch (e) {
+            // Intentional skip
+        }
+
+        const ingressInfo = getFirstIngressFromApplication(application, environment);
+
+        ms = {
+            dolittle: {
+                applicationId: application.id,
+                tenantId: application.tenantId,
+                microserviceId,
+            },
+            name: currentMicroservice.name,
+            kind: 'manual', // TODO something better?
+            environment: _props.environment,
+            extra: {
+                ingress: {
+                    path: '/todo', // TODO I think we should look this up
+                    pathType: 'Prefix',
```

Comment:
```suggestion
                    path: ingressInfo.path,
                    pathType: ingressInfo.pathType,
```

Or was there a reason why you didn't use the `ingressInfo`? Not like the view that gets this `ms` variable cares about them anymore.

Reply:
Done, good catch :)